packages/access-decision-manager-react/src/access-decision-manager-provider.tsx

```diff
@@ -0,0 +1,24 @@
+import React, {createContext, useContext} from 'react';
+import AccessDecisionManager, {Voter} from '@wizeline/access-decision-manager';
+
+interface AccessDecisionManagerContextType {
+  accessDecisionManager: AccessDecisionManager;
+}
+
+const accessDecisionManagerContext = createContext<AccessDecisionManagerContextType>({});
+
+
+const AccessDecisionManagerProvider = ({voters, children, user}: {voters: Voter[], children: any, user: any}) => {
+  const accessDecisionManager = new AccessDecisionManager(user, voters, null);
+
+  return <accessDecisionManagerContext.Provider value={accessDecisionManager}>{children}</accessDecisionManagerContext.Provider>
+};
+
+export const useIsGranted = async (attribute:any, subject: any) => {
```

Comment:
Can we add an `examples` folder with a simple Create React App that uses this?

I don't believe that hooks can be `async`, so I think there's some additional work that needs to be done here.

I think what we need to do is have a `useEffect` that watches `accessDecisionManager`, `attribute`, and `subject`, and returns that.

Maybe something like:
```jsx
const initialState = {
  error: undefined,
  isGranted: undefined,
  loading: true,
};

function reducer(state, action) {
  switch (action.type) {
    case 'request':
      return initialState;
    case 'response':
      return {
        error: undefined,
        isGranted: action.isGranted,
        loading: false,
      };
    case 'error':
      return {
        error: action.error,
        isGranted: undefined,
        loading: false,
      };
    default:
      throw new Error();
  }
}

export const useIsGranted = (attribute:any, subject: any) => {
  const  accessDecisionManager = useContext(accessDecisionManagerContext);
  const [state, dispatch] = useReducer(reducer, initialState);
  useEffect
    (() => {
      dispatch({ type: 'request'});

      accessDecisionManager.isGranted(attribute, subject)
        .then((isGranted) => dispatch({ type: 'response', isGranted}))
        .then((error) => dispatch({ type: 'response', error}))
    },
    [accessDecisionManager, attribute, subject]
  );

  return state
};
```

Reply:
Thanks a lot! you are totally right I was having problems trying to handle the promise but that approach solves the issue. I just pushed a new commit b41970ddd9c3d13b1395ac52f329bd2e3bc26a1d. the next ones will be for testing and examples